lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -368,12 +405,29 @@ export class UserAgentApplication {
         // validate request
         const request: AuthenticationParameters = RequestUtils.validateRequest(userRequest, false, this.clientId, Constants.interactionTypePopup);
 
+        const requestCorrelationId = userRequest.correlationId || CryptoUtils.createNewGuid();
+        const apiEvent: ApiEvent = new ApiEvent(requestCorrelationId, this.logger);
+        apiEvent.apiEventIdentifier = API_EVENT_IDENTIFIER.AcquireTokenPopup;
+        apiEvent.apiCode = API_CODE.AcquireTokenPopup;
+        this.telemetryManager.startEvent(apiEvent);
+
         return new Promise<AuthResponse>((resolve, reject) => {
             this.acquireTokenInteractive(Constants.interactionTypePopup, false, request, resolve, reject);
-        }).catch((error: AuthError) => {
-            this.cacheStorage.resetTempCacheItems(request.state);
-            throw error;
-        });
+        })
+            .then((resp) => {
+                apiEvent.wasSuccessful = true;
+                return resp;
+            })
+            .catch((error: AuthError) => {
+                this.cacheStorage.resetTempCacheItems(request.state);
+                apiEvent.apiErrorCode = error.errorCode;
```

Comment:
I would prefer to see this more cleanly separated from business logic, if possible.

Reply:
May be add a helper that:
- Cleans the cache
- Adds telemetry instrumentation
?

Reply:
resolved